mantidproject/mantid
# Remove runs with bad proton charges from added runs

Matthew has confirmed that this approach will do and is a close enough match to what Freddie is doing for new runs.

**Description of work.**
There is a bug which can cause SANS2D files to contain bad times in the proton charge logs (bad = the pulse occurred in the 1700s). While this has been fixed for new files being written, it can cause subtle bugs if these bad runs are included in added files.

This PR removes bad runs from added files, and predicts what the values should be.

While I was digging around the old SANS code I also fixed a few code quality issues, as the volume of squiggly yellow lines made it difficult to read any code at all.

**Report to:** Sarah SANS2D

**To test:**
1. Interfaces -> SANS -> ISIS SANS
2. Ensure ISIS data archive is on
3. Load the attached user file in order to set your instrument to SANS2D
4. Click the sum runs tab (left)
5. In the top bar, enter 58090-58095 (this will add all runs between 58090 and 58095. The bad run is 58093). Make sure the overlay checkbox is ticked and click run.
6. Load the added file back into Mantid, look at its logs and click `proton charge`. The numbers should seem sensible i.e no negative log times, or 0 proton charges

Fixes #25721


**Data:**
[user_file.txt](https://github.com/mantidproject/mantid/files/3107056/user_file.txt)

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
